rust/datafusion/src/execution/physical_plan/expressions.rs

```diff
@@ -299,6 +356,56 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn cast_i32_to_u32() -> Result<()> {
+        let schema = Schema::new(vec![Field::new("a", DataType::Int32, false)]);
+        let a = Int32Array::from(vec![1, 2, 3, 4, 5]);
+        let batch = RecordBatch::try_new(Arc::new(schema.clone()), vec![Arc::new(a)])?;
+
+        let cast = CastExpr::try_new(col(0), &schema, DataType::UInt32)?;
+        let result = cast.evaluate(&batch)?;
+        assert_eq!(result.len(), 5);
+
+        let result = result
+            .as_any()
+            .downcast_ref::<UInt32Array>()
+            .expect("failed to downcast to UInt32Array");
```

Comment:
I'm curious about how the API is designed to work here.  Why don't you handle the downcast in `evaluate` based on `data_type()`?  Is it just that you don't want the user to always incur the cost of the match on `data_type()`.